File: Aqua Guard/frontend/app/src/main/java/com/aquaguard/domain/repository/AuthRepository.java
package com.aquaguard.domain.repository;

import androidx.lifecycle.LiveData;
import com.aquaguard.domain.model.User;

public interface AuthRepository {
    String DEFAULT_ADMIN_EMAIL = "admin@aquaguard.com";
    String DEFAULT_ADMIN_PASSWORD = "Admin@123";
    LiveData<User> getCurrentUser();
    void signInWithEmail(String email, String password, RepositoryCallback<User> callback);
    void signUpWithEmail(String email, String password, String displayName, RepositoryCallback<User> callback);
    void signOut(RepositoryCallback<Void> callback);
    void updateHouseholdDetails(int familyMembers, int dailyTargetLiters, RepositoryCallback<Void> callback);
    void sendPasswordResetEmail(String email, RepositoryCallback<Void> callback);
}

File: Aqua Guard/frontend/app/src/main/java/com/aquaguard/domain/repository/AuthRepository.kt
package com.aquaguard.domain.repository

import com.aquaguard.domain.model.User
import kotlinx.coroutines.flow.Flow

interface AuthRepository {
    val currentUser: Flow<User?>
    suspend fun signInWithEmail(email: String, password: String): Result<User>
    suspend fun signUpWithEmail(email: String, password: String, displayName: String): Result<User>
    suspend fun signOut()
    suspend fun updateHouseholdDetails(familyMembers: Int, dailyTargetLiters: Int): Result<Unit>
    suspend fun sendPasswordResetEmail(email: String): Result<Unit>
}
